app/views/posts/index.html.erb

```diff
@@ -1,104 +1,17 @@
-<div class="user-list-container">
-    <div class="user-image-container">
-        X
-    </div>
-    <div class="user-detail-container">
-        <h3 class="user-name">User Name </h3>
-        <span class="num-posts">Number of posts</span>
-    </div>
-</div>
+<%= render @post[0].author %>
 
-<div class="posts-list-container">
-    <h4>Post #<%= Random.rand(1..9999) %> </h4>
-    <p class="posts-content">
-        Lorem ipsum dolor sit amet, consectetur adipiscing elit,
-        sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
-        Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris
-        nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in
-        reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
-        Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt
-        mollit anim id est laborum.
-    </p>
-    <div class="interaction-container">
-        <span class="comments-counter">Comments X</span>
-        <span class="likes-counter">Likes Y</span>
-    </div>
-</div>
+<% @post.each do |post|%>
+<%= render post %>
+<% if post.comments.length != 0 %>
 <div class="comments-container">
-    <div class="comments-list-container">
-        <span class="commenter-user-name">Username:</span>
-        <p class="comments-content">Comment <%= Random.rand(1..20) %></p>
-    </div>
-    <div class="comments-list-container">
-        <span class="commenter-user-name">Username:</span>
-        <p class="comments-content">Comment <%= Random.rand(1..20) %></p>
-    </div>
-    <div class="comments-list-container">
-        <span class="commenter-user-name">Username:</span>
-        <p class="comments-content">Comment <%= Random.rand(1..20) %></p>
-    </div>
-    <div class="comments-list-container">
-        <span class="commenter-user-name">Username:</span>
-        <p class="comments-content">Comment <%= Random.rand(1..20) %></p>
-    </div>
-
+    <%= render post.comments %>
```

Comment:
- [ ] You should use the recent comments method in the post model for this loop instead of rendering all comments.